src/resolve/resolved-dns-query.c

```diff
@@ -899,6 +901,12 @@ static void dns_query_accept(DnsQuery *q, DnsQueryCandidate *c) {
                         DNS_ANSWER_REPLACE(q->answer, dns_answer_ref(t->answer));
                         q->answer_rcode = t->answer_rcode;
                         q->answer_dnssec_result = t->answer_dnssec_result;
+                        q->answer_ede_rcode = t->answer_ede_rcode;
+                        if (t->answer_ede_msg) {
+                                q->answer_ede_msg = strdup(t->answer_ede_msg);
+                                if (!q->answer_ede_msg)
+                                        log_oom();
```

Comment:
oom should really be fatal in almost all cases. hence please do `log_oom()` followed by `goto fail` here. See comment above though.